### PR TITLE
Azure Monitor Exporter: Service Version

### PR DESCRIPTION
**Pull request by TimothyMothra**
Addressing #18081 

See also:
- [swagger definition](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/applicationinsights/data-plane/Monitor.Exporters/preview/2020-09-15_Preview/swagger.json#L4)
- [dotnet design guidelines: service version](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-service-version-option)

TODO
- ~Need to discuss with Pavel and Srikanta about the correct way to pass this value into the generated RestClient~ Deferring this task until there is a second service version.

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorExporterOptions.cs [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
+
 using Azure.Core;
 
 namespace Azure.Monitor.OpenTelemetry.Exporter
 {
     public class AzureMonitorExporterOptions : ClientOptions
     {
         public string ConnectionString { get; set; }
+
+        [EditorBrowsable(EditorBrowsableState.Never)]
```
Comment (pakrym) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorExporterOptions.cs:14]: Don't add this property.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/ServiceVersion.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+using System.Diagnostics.CodeAnalysis;
+
+namespace Azure.Monitor.OpenTelemetry.Exporter
+{
+    /// <summary>
+    /// These values define the version of the Azure Monitor Ingestion Service.
+    /// </summary>
+    [SuppressMessage("Naming", "CA1707:Identifiers should not contain underscores", Justification = "This naming format is defined in the Azure SDK Design Guidelines")]
+    public enum ServiceVersion
```
Comment (pakrym) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/ServiceVersion.cs:12]: This class should be nested into client options see https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClientOptions.cs#L23

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorExporterOptions.cs [hunk 3]
```diff
@@ -1,12 +1,17 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using System.ComponentModel;
+
 using Azure.Core;
 
 namespace Azure.Monitor.OpenTelemetry.Exporter
 {
     public class AzureMonitorExporterOptions : ClientOptions
```
Comment (pakrym) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorExporterOptions.cs:10]: Options should define a ctor that takes ServiceVersion as a defaulted parameter: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClientOptions.cs#L47

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorExporterOptions.cs (hunk 1 above)
Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorExporterOptions.cs:14]: Do we want to expose this property if it's not in use yet?

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/ServiceVersion.cs (hunk 2 above)
Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/ServiceVersion.cs:12]: oh okay, that's an easy change :)

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorExporterOptions.cs (hunk 1 above)
Comment (pakrym) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorExporterOptions.cs:14]: Nope, removing it is fine.

**Review by pakrym — APPROVED** [review]

**Pull request merged**